src/main/java/software/amazon/awssdk/crt/http/HttpClientConnection.java

```diff
@@ -63,9 +63,12 @@ protected HttpClientConnection(HttpClientConnectionManager manager, long connect
                     streamHandler);
                 if (stream == null || stream.isNull()) {
                     streamFuture.completeExceptionally(new RuntimeException("HttpStream creation failed"));
+                    return streamFuture;
                 }
+
+                stream.activate();
                 streamFuture.complete(stream);
```

Comment:
How is automatically calling activate() any different from not having activate() at all?

Explicit activate() is to solve the issue where people write code where they do something with the returned stream (ex: store it as a member variable somewhere), then they expect to be able to access the stream via that member variable from the callbacks.

If we automatically call activate() then yes, we've solved our internal implementation issues, but anyone who wraps our stuff in their own system will still have these issues.

On a side note, I have no idea why this function returns a CompletableFuture<HttpStream> when it could just return an HttpStream.

Reply:
not done any more, and no longer returning a future